### ISSUE TITLE
Allow Terraform Providers to Configure Issue Tracker Information and Return in Error Messaging

### ISSUE THREAD

**Issue by bflad** [hashicorp/terraform-plugin-sdk#140]
### Current Terraform Version

```
Terraform v0.12.0-beta1
```

### Use-cases

Some of Terraform's error messaging suggests creating an issue in a Terraform Provider's issue tracking:

```
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker.
```

The location of this issue tracker may be ambiguous or not necessarily known by operators, especially for third-party Terraform Providers.

### Proposal

It would be great to allow Terraform Providers to configure a new string value in their schema which can be inserted in these locations, such as:

```
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker: https://github.com/terraform-providers/terraform-provider-aws/issues
```

I hesitate to say _URL_ specifically because some third-party providers may prefer:

```
        This is a bug in the provider, which should be reported in the provider's own
        issue tracker: helpdesk@example.com
```

We can still fall back to the old message (ending with `issue tracker.`) if the new provider configuration is not present.

**Comment by paultyng** [comment]
I think this really should be a registry concern as well, if the provider was sourced from the registry, its support information can be funneled back through the registry to the appropriate location. The CLI can infer the registry landing page in 0.12 I believe, and that can probably be the appropriate landing page for support routing. Hard coding this in the provider will just cause drift and potential issues actually finding the proper reporting location (not to mention impacts of forking code, etc).

**Comment by paultyng** [comment]
The registry already has this button:

![image](https://user-images.githubusercontent.com/422474/72836108-fd70b780-3c59-11ea-8104-82eab76556e6.png)

**Comment by radeksimko** [comment]
Agreed, that's a great idea @paultyng We could have a canonical URL, such as `https://registry.terraform.io/providers/hashicorp/aws/latest/issue` which will redirect to GitHub or whatever else is set in the Registry.

With that said:

1. this redirect is not exposed from the Registry yet - so we should probably create a relevant task for the Registry team to get this done
2. When it is exposed we still need to implement this somehow into the SDK and/or core.

So while this issue may not represent _how_ we are going to solve it, it still represents the unsolved problem.

For that reason I think we should keep it open until we have a replacement issue to link to.
